tools/continuous_integration/bisect.py

```diff
@@ -0,0 +1,104 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script to bisect the mainline commits and find the culprit commit.
+# The default 'git bisect' checks feature branches, which is not desired
+# because commits in feature branch might not pass tests or compile.
+#
+#   python bisect.py <git-repo-dir> <build-dir> \
+#       <last-good-commit> <bad-commit> <test-target> 
+#
+# Update BRANCH for non develop branch. Update REPEAT for very flaky test.
+# Update PARALLEL for build paralleism.
+#
+# Example:
+#   python bisect.py $PWD/../Paddle2 $PWD e80059 6bba41 test_rnn_encoder_decoder
```

Comment:
Can we remove the first command line parameter (the git repo dir) by looking for the nearest ancestor directory of $PWD which has a `.git` subdirectory?

Reply:
I sometimes create the build directory in a different directory outside the git repo. That way, I can mount an in-memory filesystem for faster build.